packages/block-library/src/site-logo/index.php

```diff
@@ -111,54 +111,52 @@ function _override_custom_logo_theme_mod( $custom_logo ) {
 /**
  * Updates the site_logo option when the custom_logo theme-mod gets updated.
  *
- * This function is hooked on "update_option_theme_mods_$theme" and not
- * "pre_set_theme_mod_custom_logo" because by hooking in `update_option`
- * the function accounts for remove_theme_mod() as well.
- *
- * @param mixed $old_value The old option value.
- * @param mixed $value     The new option value.
+ * @param  mixed $value Attachment ID of the custom logo or an empty value.
+ * @return mixed
  */
-function _sync_custom_logo_to_site_logo( $old_value, $value ) {
-	// Delete the option when the custom logo does not exist or was removed.
-	// This step ensures the option stays in sync.
-	if ( empty( $value['custom_logo'] ) ) {
+function _sync_custom_logo_to_site_logo( $value ) {
+	if ( empty( $value ) ) {
 		delete_option( 'site_logo' );
 	} else {
-		remove_action( 'update_option_site_logo', '_sync_site_logo_to_custom_logo' );
-		update_option( 'site_logo', $value['custom_logo'] );
-		add_action( 'update_option_site_logo', '_sync_site_logo_to_custom_logo', 10, 2 );
+		update_option( 'site_logo', $value );
 	}
+
+	return $value;
 }
 
+add_filter( 'pre_set_theme_mod_custom_logo', '_sync_custom_logo_to_site_logo' );
```

Comment:
I'm restoring the use of the `pre_set_theme_mod_custom_logo` filter because it solves a couple problems that happen when switching to the `update_option_theme_mods_$theme` action in #32370.

- Make sure the site_logo isn't deleted in the case where a theme isn't setting a logo because it doesn't support one
- Prevent the site_logo from being deleted when switching themes.